components/script/dom/xmlhttprequest.rs

```diff
@@ -979,14 +979,12 @@ impl XMLHttpRequest {
                 // Part of step 11, send() (processing response end of file)
                 // XXXManishearth handle errors, if any (substep 2)
 
-                // Subsubsteps 5-7
+                // Subsubsteps 6-8
                 self.send_flag.set(false);
 
                 self.change_ready_state(XMLHttpRequestState::Done);
                 return_if_fetch_was_terminated!();
-                // Subsubsteps 10-12
-                self.dispatch_response_progress_event(atom!("progress"));
```

Comment:
It's intentional that subsubstep 5 is not included. This comment https://github.com/whatwg/xhr/issues/72#issuecomment-247321220 clearly states that the ```e.loaded``` value should never decrease or be equals to the last value.

In the Servo implementation of XHR, the ```XHRProgress::Done``` progress event is only passed when we finished the send meaning the last ```e.loaded``` value was the total byte send. Firing the subsubstep 5 will gave us the same ```e.loaded``` value. 

The idea of comment https://github.com/whatwg/xhr/issues/72#issuecomment-247182864 is to not send a ```progress``` event when ```e.loaded == e.total``` in the ```XHRProgress::Loading```. The big problem is that ```e.total``` could be 0 (example with ```progress-events-response-data-gzip.htm``` test case).

I don't really know what I should do? Should if file a bug at https://github.com/whatwg/xhr?